src/staking/index.ts

```diff
@@ -0,0 +1,121 @@
+import { StakingPoolFactory__factory, StakingPool__factory } from "@energyweb/iam-contracts";
+import { StakingPool } from "@energyweb/iam-contracts/dist/ethers-v4/StakingPool";
+import { StakingPoolFactory } from "@energyweb/iam-contracts/dist/ethers-v4/StakingPoolFactory";
+import { utils } from "ethers";
+import { BigNumber } from "ethers/utils";
+import { ERROR_MESSAGES } from "../errors";
+import { IAM } from "../iam";
+import { chainConfigs } from "../iam/chainConfig";
+import { EncodedCall } from "../iam/iam-base";
+
+const { namehash } = utils;
+/**
+ * @interface
+ */
+export interface IStakable {
```

Comment:
Some initial description on what the interface is for would be nice if you have time. Maybe something like below?
```suggestion
/**
 * @interface
 * @description Provides functionality to launch and interact with staking pools
 */
export interface IStakable {
```